src/cmd/cmd_conc.ml

```diff
@@ -0,0 +1,240 @@
+  (*       let memories = Thread.memories t in *)
+  (*       Symbolic_memory.iter *)
+  (*         (fun tbl -> *)
+  (*           Symbolic_memory.ITbl.iter *)
+  (*             (fun _ (m : Symbolic_memory.t) -> *)
+  (*               Symbolic_memory.replace_size m base size ) *)
+  (*             tbl ) *)
+  (*         memories; *)
+  (*       Expr.make (Ptr (base, Value.const_i32 0l)) ) *)
+  (* in *)
+  (* let free (p : Value.int32) : unit Choice.t = *)
+  (*   let base = ptr p in *)
+  (*   Choice.with_thread (fun t -> *)
+  (*       let memories = Thread.memories t in *)
+  (*       Symbolic_memory.iter *)
+  (*         (fun tbl -> *)
+  (*           Symbolic_memory.ITbl.iter *)
+  (*             (fun _ (m : Symbolic_memory.t) -> Symbolic_memory.free m base) *)
+  (*             tbl ) *)
+  (*         memories ) *)
+  (* in *)
+  let functions =
+    [ (*   ( "alloc" *)
+      (*   , Symbolic.P.Extern_func.Extern_func *)
+      (*       (Func (Arg (I32, Arg (I32, Res)), R1 I32), alloc) ) *)
+      (* ; ( "dealloc" *)
+      (*   , Symbolic.P.Extern_func.Extern_func (Func (Arg (I32, Res), R0), free) ) *)
+      (* ; ("abort", Symbolic.P.Extern_func.Extern_func (Func (UArg Res, R0), abort)) *) ]
+  in
+  { functions }
+
+let ( let*/ ) (t : 'a Result.t) (f : 'a -> 'b Result.t Choice.t) :
+  'b Result.t Choice.t =
+  match t with Error e -> Choice.return (Error e) | Ok x -> f x
+
+let simplify_then_link_then_run ~unsafe ~optimize (pc : unit Result.t Choice.t)
+  (m : Text.modul) =
+  let link_state = Link.empty_state in
+  let link_state =
+    Link.extern_module' link_state ~name:"symbolic"
+      ~func_typ:Concolic.P.Extern_func.extern_type symbolic_extern_module
+  in
+  let link_state =
+    Link.extern_module' link_state ~name:"summaries"
+      ~func_typ:Concolic.P.Extern_func.extern_type summaries_extern_module
+  in
+  let*/ to_run, link_state =
+    let has_start =
+      List.exists (function Text.MStart _ -> true | _ -> false) m.fields
+    in
+    let has_start_id_function =
+      List.exists
+        (function Text.MFunc { id = Some "_start"; _ } -> true | _ -> false)
+        m.fields
+    in
+    let fields =
+      if has_start || not has_start_id_function then m.fields
+      else MStart (Text "_start") :: m.fields
+    in
+    let m = { m with fields } in
+    let+ m, state =
+      Compile.until_link ~unsafe link_state ~optimize ~name:None m
+    in
+    let m = Concolic.convert_module_to_run m in
+    (m, state)
+  in
+  let link_state_envs =
+    (* TODO: fix (move globals/memories/tables back to thread and fix this)
+       Or we can fix this by sharing the result of Concolic.convert_env
+    *)
+    Env_id.map Concolic.convert_env link_state.envs
+  in
+  let c = (Interpret.Concolic.modul link_state_envs) to_run in
+  Choice.bind pc (fun r ->
+      match r with Error _ -> Choice.return r | Ok () -> c )
+
+let run_file ~unsafe ~optimize pc filename =
+  let*/ m0dule = Parse.Module.from_file filename in
```

Comment:
I believe we took the `modul` convention (although I have to admit `m0dule` is funny).